suitcss/suit
# Convert to monorepo using lerna.js

@simonsmith @giuseppeg 

* Moves suit to a monorepo using lerna.js. 
* imports all non-deprecated packages with their commit history.
* Bootstraps all packages using the yarn client. 
* Updates mocha in generator-suit and removes pre publish`nsp` (bc of reported security issue, `lerna bootstrap` would not finish otherwise). 
* Updates uris for `repo`, `homepage` and `bugs` in all packages.

Use 
* `lerna bootstrap` to install all packages

To test lerna versioning
* Use `lerna version patch --yes --no-push` to test locally before doing anything on remote. 

To publish
* `npm login`, then
* `lerna publish patch --yes`

This results in the following version bumps, everything should stay 100% backwards compatible. 

```
 - suitcss-base: 5.0.0 => 5.0.1
 - suitcss-components-arrange: 1.1.1 => 1.1.2
 - suitcss-components-button: 6.0.2 => 6.0.3
 - suitcss-components-flex-embed: 2.0.4 => 2.0.5
 - suitcss-components-grid: 4.0.0 => 4.0.1
 - suitcss-components-test: 1.0.0 => 1.0.1
 - suitcss-components: 2.0.0 => 2.0.1
 - generator-suit: 1.0.2 => 1.0.3
 - suitcss-preprocessor: 4.0.0 => 4.0.1
 - stylelint-config-suitcss: 14.0.0 => 14.0.1
 - stylelint-suitcss: 3.0.0 => 3.0.1
 - suitcss-theme: 1.0.0 => 1.0.1
 - suitcss-utils-after: 1.0.1 => 1.0.2
 - suitcss-utils-align: 1.0.0 => 1.0.1
 - suitcss-utils-before: 1.0.1 => 1.0.2
 - suitcss-utils-display: 1.0.2 => 1.0.3
 - suitcss-utils-flex: 2.0.1 => 2.0.2
 - suitcss-utils-layout: 1.0.1 => 1.0.2
 - suitcss-utils-link: 1.0.1 => 1.0.2
 - suitcss-utils-offset: 1.0.0 => 1.0.1
 - suitcss-utils-position: 1.0.1 => 1.0.2
 - suitcss-utils-size: 2.0.1 => 2.0.2
 - suitcss-utils-text: 1.0.0 => 1.0.1
 - suitcss-utils: 3.0.0 => 3.0.1
```

This PR is limited in scope intentionally and should be a good foundation for further development work with a monorepo approach. 

#151
